internal/monitors/device-updater/device_updater.go

```diff
@@ -47,42 +50,119 @@ func (d *DeviceUpdater) UpdateDevices() {
 		return
 	}
 
-	for _, fleet := range fleets {
+	for fleetIndex := range fleets {
+		orgID := fleets[fleetIndex].OrgID
 		// If there is no template set in the fleet, then there is nothing to sync to the devices
-		if fleet.Spec == nil || fleet.Spec.Data.Template.Metadata == nil {
+		if fleets[fleetIndex].Spec == nil || fleets[fleetIndex].Spec.Data.Template.Metadata == nil {
 			continue
 		}
 
-		devices, err := d.devStore.ListIgnoreOrg(fleet.Spec.Data.Selector.MatchLabels)
+		listParams := service.ListParams{
+			Labels: fleets[fleetIndex].Spec.Data.Selector.MatchLabels,
+			Limit:  service.MaxRecordsPerListRequest, //TODO: Handle paging
+		}
+		devices, err := d.devStore.List(ctx, orgID, listParams)
 		if err != nil {
-			log.Errorf("failed to list devices for fleet %s: %v", fleet.Name, err)
+			log.Errorf("failed to list devices for fleet %s: %v", fleets[fleetIndex].Name, err)
 		}
 
-		for _, device := range devices {
-			updated := d.updateDeviceSpecAccordingToFleetTemplate(log, &device, &fleet) //nolint:gosec
+		for devIndex := range devices.Items {
+			dev := devices.Items[devIndex]
+			if dev.Metadata.Owner == nil || *dev.Metadata.Owner == "" {
+				dev.Metadata.Owner = util.SetResourceOwner(model.FleetKind, fleets[fleetIndex].Name)
+				log.Infof("Setting previously unset fleet of device %s to %s", *dev.Metadata.Name, *dev.Metadata.Owner)
+			} else if *dev.Metadata.Owner != fleets[fleetIndex].Name {
+				// The fleet for this device changed
+				err = d.handleOwningFleetChanged(ctx, log, &dev, &fleets[fleetIndex])
+				if err != nil {
+					log.Errorf("failed to handle fleet update for device %s: %v", *dev.Metadata.Name, err)
+				}
```

Comment:
Can you help me understand why UpdateDevices() doesn't return an error. You could aggregate errors but in terms of testability we would need to parse logs to validate failure?